src/browser/KommunicateCordovaPluginProxy.js

```diff
@@ -0,0 +1,302 @@
+		console.log("Received object : " + kmUser);
+		if (isUserLoggedIn() && typeof Kommunicate != 'undefined' && Kommunicate) {
+			successCallback("success")
+		} else {
+			initPlugin(JSON.parse(kmUser), successCallback, errorCallback)
+		}
+	},
+
+	isLoggedIn: function (successCallback, errorCallback) {
+		console.log("Called isLoggedIn function ");
+		successCallback(isUserLoggedIn() ? "true" : "false")
+	},
+
+	launchConversation: function (successCallback, errorCallback) {
+		console.log("Called launchConversation function ");
+		init((response) => {
+			Kommunicate.launchConversation();
+			parent.document.getElementById('kommunicate-widget-iframe').setAttribute("style", "display:block");
+			successCallback(response)
+		}, (error) => {
+			errorCallback(error)
+		});
+	},
+
+	launchParticularConversation: function (successCallback, errorCallback, conversationObj) {
+		console.log("Called launchParticularConversation function : " + JSON.parse(conversationObj).clientChannelKey);
+		init((response) => {
+      KommunicateGlobal.document.getElementById("mck-sidebox-launcher").click();
+			KommunicateGlobal.$applozic.fn.applozic('loadGroupTabByClientGroupId', {
+				"clientGroupId": JSON.parse(conversationObj).clientChannelKey
+			});
+			parent.document.getElementById('kommunicate-widget-iframe').setAttribute("style", "display:block");
+			successCallback(response)
+		}, (error) => {
+			errorCallback(error)
+		});
+	},
+
+	conversationBuilder: function (successCallback, errorCallback, conversationObjectStr) {
+		console.log("Called conversationBuilder function : " + conversationObjectStr);
+		var conversationObj = JSON.parse(conversationObjectStr);
+		var kmUser = {};
+
+		if (isUserLoggedIn()) {
+			init((response) => {
+				createConversation(conversationObj, JSON.parse(localStorage.KM_PLUGIN_USER_DETAILS).userId, successCallback, errorCallback);
+			}, (error) => {
+				errorCallback(error)
+			})
+		} else {
+			if (conversationObj.kmUser) {
+				kmUser = JSON.parse(conversationObj.kmUser);
+				kmUser.applicationId = conversationObj.appId
+			} else if (conversationObj.withPreChat && conversationObj.withPreChat == true) {
+				kmUser.withPreChat = true;
+        kmUser.applicationId = conversationObj.appId
+			} else {
+				kmUser = {
+					'userId': getRandomId(),
+					'applicationId': conversationObj.appId
+				}
+			}
+			initPlugin(kmUser, (response) => {
+        if(!(kmUser.withPreChat && kmUser.withPreChat == true)) {
+         createConversation(conversationObj, kmUser.userId, successCallback, errorCallback);
+        }
+			}, (error) => {
+				errorCallback(error);
+			});
+		}
+	},
+
+	logout: function (successCallback, errorCallback) {
+		console.log("Called logout function ");
+		if (isUserLoggedIn() && typeof Kommunicate != 'undefined' && Kommunicate) {
+			init((response) => {
+				//Kommunicate.launchConversation();
+				Kommunicate.logout();
+				localStorage.removeItem('KM_PLUGIN_USER_DETAILS');
+				location.reload()
```

Comment:
@reytum is this required?

Reply:
location.reload can be removed.

Reply:
Its used to clear the Kommunicate variable as if someone logs in just after logging out the error comes : script already loaded.